meta-balena-common/recipes-core/initrdscripts/files/migrate

```diff
@@ -144,14 +195,14 @@ migrate_run() {
         _config_json_name=$(basename "${CONFIG_PATH}")
         cp "${FLASH_BOOT_MOUNT}/${_config_json_name}" "${CONFIG_PATH}"
 
-        umount "${FLASH_BOOT_MOUNT}" > /dev/null || true
+        disable_persistent_logging
+        while ! umount "${FLASH_BOOT_MOUNT}"; do continue; done
```

Comment:
@jakogut if the storage is blocked and cannot be unmounted this will never end leaving the device unreachable. I'd rather try to umount once and on failure do a lazy umount.

Reply:
That seems like an acceptable compromise. I changed this to requiring an unmount for the boot partition as I noticed that an expected failed flashing attempt (OOM) left the FAT32 boot partition unclean.

Reply:
@jakogut what did you decide here? Are you leaving the infinite umount loop or opt for a sequence of umounts and a final lazy umount to avoid the possibility of blocking boot here?

Reply:
@alexgg I think, given the problems I encountered with overwriting the boot partition at the block level while it was mounted, we shouldn't proceed with the migrate if it remains mounted. We won't be able to trust the integrity of the installed system after the fact.

If the unmount fails, is rebooting a safer option?

Reply:
hey @jakogut I agree with not going ahead if the boot partition does not umount, but I don't want an infinite loop either in the migration code. Something like:
```
if ! timeout 10 bash -c 'until umount ${FLASH_BOOT_MOUNT}; do sleep 1; done'; then
    info "Failed to umount ${FLASH_BOOT_MOUNT}"
    exit 1
fi
```

Reply:
@alexgg I like that approach, we'll do that.